### PR TITLE
Ported /tg/'s outfit datums, refactored job controller to support them

### PR DESCRIPTION
**Pull request by tigercat2000**
Players:
 - The only thing that you should notice is that your backpack options
 have changed. Instead of being "None", "Backpack", "Satchel", and
 "Satchel Alt", there are 6 options:
   - "Grey Backpack": Ensures you always recieve a plain grey backpack.
   - "Grey Satchel": Same as the above, grey, plain, satchel.
   - "Grey Dufflebag": So you don't have to rush the locker room, I
   guess. Just kinda ransacked it from /tg/. Happy to revert it, please
   do not post a five paragraph text wall calling me a shitlord
   powergaming hitler-loving furry.
   - "Leather Satchel": It's a leather satchel. What the hell did you
   think it was?
   - "Department Backpack": Gives you the department's default backpack.
   Grey for most jobs.
   - "Department Satchel": It gives you the department's default
   satchel; A grey satchel if the department doesn't have one.

Coders:
 - The above means this needs a minor SQL change:
   ```sql
   ALTER TABLE characters MODIFY backbag MEDIUMTEXT;
   UPDATE characters SET backbag='Grey Backpack';
   ```

 - This is a cleaner method of handling job equipment than
 "M.equip_or_collect" x 1000. Jobs specify a '/datum/outfit' datum,
 which defines all of the things to equip to the user.

 - Minor note: equip_or_collect will no longer ever generate a plastic
 bag. It will instead just drop stuff on the floor if it can't stuff it
 into a mob. If this is even slightly a point of contention, more than
 happy to revert it; It was just requested by Fox.
   - Addendum: This also has the effect of making it so that labcoats
   are just thrown onto the floor if you have a loadout suit. It's not
   really avoidable. Uniforms and suits are equipped before the
   backpack.

 - Changes from /tg/:
   - Changed all of the "H.equip_to_slot_or_del" calls into invoking a
   new "equip_item" proc; This is used to support a new variable called
   "collect_not_del". Job outfits use this, it makes it so that the
   system calls equip_or_collect instead of equip_to_slot_or_del. The
   idea being that job items should never be lost to the void,
   especially because they can conflict with loadout items.

TODO:
 - [x] Retrofit "Equip Mob" verb to use this for all of the non-job options
 - [x] Fully implement head announcements(?) (Yeah I'll do this seperately, no need to shove it in here)
   - [x] May be best for a second PR that addresses all of the real issues
   with our current job system

### CODE REVIEW

**Comment by KasparoVy** [comment]
Wow, nice. Great work

**Comment by Markolie** [comment]
We should probably also apply this (if possible) to the select equipment command and things such as ERT uniforms. 

Good stuff!

**Comment by tigercat2000** [comment]
@Markolie 
```

**Comment by Markolie** [comment]
Welp!

**Comment by ModDownloading** [comment]
No department dufflebags? If we are doing the gray ones, I would appreciate those.

**Comment by fludd12** [comment]
Woe to my poor unsung hero, the plastic bag - the civilian's discount toolbelt. The medical scientist's replacement pillbottle!

We hardly knew ye.

**Comment by taukausanake** [comment]
I think plastic bags need more work to be used. For starters a way to get them in-game. Also something like double-bagging to carry more and more heavy. But if you go over that limit the bag(s) tear and everything falls out. I tried messing with it once on my own but I was really fucking bad at making it work

**Comment by fludd12** [comment]
Oh, they can be made with plastic sheets, if I recall correctly. I was just being heavily melodramatic.

File: code/_globalvars/lists/flavor_misc.dm [hunk 1]
```diff
@@ -46,4 +44,13 @@ var/list/TAGGERLOCATIONS = list("Disposals",
...
\ No newline at end of file
+var/list/hit_appends = list("-OOF", "-ACK", "-UGH", "-HRNK", "-HURGH", "-GLORF")
+
+	//Backpacks
+#define GBACKPACK "Grey Backpack"
+#define GSATCHEL "Grey Satchel"
+#define GDUFFLEBAG "Grey Dufflebag"
+#define LSATCHEL "Leather Satchel"
+#define DBACKPACK "Department Backpack"
+#define DSATCHEL "Department Satchel"
```
Comment (Fox-McCloud) [inline code/_globalvars/lists/flavor_misc.dm]: I'd go ahead and add departmental dufflebags here, too; we have a number of them already; the sprites can be ported at a later time.

**Comment by davidchan13** [comment]
As long as duffel bags still have the slowdown I don't see a problem with being able to spawn with them, department or otherwise. Odds are you'd have to carry an internals box or two full of coffee anyways just to keep your speed up to normal it's hardly worth the effort to most people.

**Comment by Fox-McCloud** [comment]
LGTM once the departmental dufflebags option is added and merge conflicts are resolved.

**Comment by tigercat2000** [comment]
@Fox-McCloud Still gotta do the select equipment stuff

**Comment by tigercat2000** [comment]
This is ready.

**Comment by KasparoVy** [comment]
:ok_hand:

**Comment by Fox-McCloud** [comment]
Merge conflicts.

**Pull request merged**